app/assets/src/components/ui/controls/SampleTypeSearchBox.jsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import { groupBy } from "lodash/fp";
+
+import PropTypes from "~/components/utils/propTypes";
+
+import LiveSearchPopBox from "./LiveSearchPopBox";
+
+class SampleTypeSearchBox extends React.Component {
+  SUGGESTED = "SUGGESTED";
+  DONOTSHOW = "DONOTSHOW";
+  ALL = "ALL";
+
+  handleSearchTriggered = query => {
+    const matchSampleTypes = sampleType => {
+      // Match chars in any position. Good for acronyms. Ignore spaces.
+      const noSpaces = query.replace(/\s*/gi, "");
+      const regex = new RegExp(noSpaces.split("").join(".*"), "gi");
+      if (regex.test(sampleType.name)) {
+        return true;
+      }
+      return false;
+    };
+    const matchedSampleTypes = this.props.sampleTypes.filter(matchSampleTypes);
+
+    const isSuggested = sampleType => {
+      // Insects only get their own types
+      if (this.props.isInsect) {
+        return sampleType.insect_only ? this.SUGGESTED : this.DONOTSHOW;
+      }
+      // Humans get any type
+      if (this.props.isHuman) {
+        return this.SUGGESTED;
```

Comment:
Nit: Could we move the constants above the class and just have `SUGGESTED` instead of `this.SUGGESTED`?

Reply:
Good point. I guess that is more idiomatic.